rs-bao/src/components/coaches/CoachLoginForm.vue

```diff
@@ -0,0 +1,209 @@
+<template>
+  <!-- First Dialog -->
+  <coach-dialog :show="loading" title="Authenticating...">
+    <coach-loading></coach-loading>
+  </coach-dialog>
+
+  <!-- Second Dialog -->
+  <coach-dialog
+    :show="!!error"
+    title="An error occurred"
+    @close="closeErrorForm"
+    isChecking
+  >
+    <p>{{ error }}</p>
+  </coach-dialog>
+
+  <form @submit.prevent="checkClick == 1 ? onLogin() : onRegister()">
+    <div class="input-group">
+      <label for="email">E-Mail</label>
+      <input type="email" name="email" id="email" v-model="email" />
+    </div>
+    <div class="input-group">
+      <label for="password">Password</label>
+      <input type="password" name="password" id="password" v-model="password" />
+    </div>
+    <p v-if="message">
+      Please enter a valid email and password (must be at least 6 characters
+      long).
+    </p>
+    <div class="button-group">
+      <div v-if="checkClick">
+        <button type="submit" id="btn-login" v-if="checkClick == 1">
+          Login
+        </button>
+        <button type="submit" id="btn-login" v-else>
+          Sign Up
+        </button>
+        <button
+          type="button"
+          @click.prevent="onChangeClick"
+          id="btn-signup"
+          v-if="checkClick == 1"
+        >
+          Signup instead
+        </button>
+        <button
+          type="button"
+          @click.prevent="onChangeClickTwo"
+          id="btn-signup"
+          v-else
+        >
+          Login instead
+        </button>
+      </div>
+    </div>
+  </form>
+</template>
+<script>
+import CoachDialog from "../common/CoachDialog.vue";
+import CoachLoading from "../common/CoachLoading.vue";
+// import firebase from "firebase/app";
+// import "firebase/auth";
+// import mapMutations from "vuex";
+
+export default {
+  components: { CoachDialog, CoachLoading },
+  name: "CoachLoginForm",
+  data() {
+    return {
+      email: "",
+      password: "",
+      checkClick: 1,
+      message: false,
+      error: null,
+      loading: false,
+    };
+  },
+  computed: {
+    user() {
+      return this.$store.getters.user;
+    },
+  },
+  watch: {
+    user(value) {
+      if (value !== null && value !== undefined) {
+        this.$router.push("/login");
+      }
+    },
+  },
+  methods: {
+    onChangeClick() {
+      return (this.checkClick = 2);
+    },
+    onChangeClickTwo() {
```

Comment:
onChangeClickTwo và onChangeClick chỉ khác nhau là set giá trị thay vì chia thành 2 thì làm theo cách này gọn hơn.
onChangeClick(number){ this.checkClick = number }